tests/test_custom_config.py

```diff
@@ -101,29 +99,58 @@ def test_cli_trumps_config(testdir, single_decorated_test_function):
                                                              test_name=test_name_exp))
     config = \
 """
-[pytest]
-ci-environment=asc
-"""  # noqa
+{
+  "pytest_zigzag_env_vars": {
+    "FOO": "foo",
+    "BAR": "bar",
+    "BUILD_URL": "foo",
+    "BUILD_NUMBER": null
+  }
+}
+""" # noqa
+
+    result = run_and_parse_with_config(testdir, config)
+
+    # Test
+    assert result[0].testsuite_props['FOO'] == 'foo'
+    assert result[0].testsuite_props['BAR'] == 'bar'
+
 
-    result_path = testdir.tmpdir.join('junit.xml')
-    config_path = testdir.tmpdir.join('conf.conf')
-    with open(str(config_path), 'w') as f:
-        f.write(config)
+def test_custom_config_precidence(testdir, single_decorated_test_function):
```

Comment:
The name of this test lead me to believe that it was going to inject a value in the JSON_config and in the ENV and see it the correct one was used.

Reply:
This is testing that when a config file path is specified both on the command line and in an ini file, that the command line option wins.  The test for an environment var taking precedence over a config file directive, is the test I've just added.